raiden/message_handler.py

```diff
@@ -125,22 +125,26 @@ def handle_message_processed(raiden: RaidenService, message: Processed):
 def on_message(raiden: RaidenService, message: Message):
     """ Return True if the message is known. """
     # pylint: disable=unidiomatic-typecheck
-    if type(message) == SecretRequest:
-        handle_message_secretrequest(raiden, message)
-    elif type(message) == RevealSecret:
-        handle_message_revealsecret(raiden, message)
-    elif type(message) == Secret:
-        handle_message_secret(raiden, message)
-    elif type(message) == DirectTransfer:
-        handle_message_directtransfer(raiden, message)
-    elif type(message) == RefundTransfer:
-        handle_message_refundtransfer(raiden, message)
-    elif type(message) == LockedTransfer:
-        handle_message_lockedtransfer(raiden, message)
-    elif type(message) == Processed:
-        handle_message_processed(raiden, message)
-    else:
-        log.error('Unknown message cmdid {}'.format(message.cmdid))
+    try:
+        if type(message) == SecretRequest:
+            handle_message_secretrequest(raiden, message)
+        elif type(message) == RevealSecret:
+            handle_message_revealsecret(raiden, message)
+        elif type(message) == Secret:
+            handle_message_secret(raiden, message)
+        elif type(message) == DirectTransfer:
+            handle_message_directtransfer(raiden, message)
+        elif type(message) == RefundTransfer:
+            handle_message_refundtransfer(raiden, message)
+        elif type(message) == LockedTransfer:
+            handle_message_lockedtransfer(raiden, message)
+        elif type(message) == Processed:
+            handle_message_processed(raiden, message)
+        else:
+            log.error('Unknown message cmdid {}'.format(message.cmdid))
+            return False
+    except Exception as e:
```

Comment:
this shouldn't be a pokemon exception but rather a specific one

Reply:
Yes, I thought about that, but I'd argue this is exactly the only place where a pokemon exception handling is valid: there's a whole monster of codepaths under state changes handling, and we'd not like Raiden crashing because of a malformed or malicious message (`on_message` is called only by the transports when a message is received), as it's a DoS vector, that's why I think this is the only place where it can be accepted.

Reply:
Then we should add some message validation and checks in the state machine - if the message is malformed, passes basic message validation and causes a crash (or even worse, creates an inconsistent state), then we should know about it and fix it ASAP.

Reply:
I'd just  remove this catch and address it in a separate issue.
Having a test that feeds malformed messages to the app would be a nice thing to have too ;)

Reply:
There's basic message validation in the transport, but valid message may still raise an exception, if the state over which it'd be applied didn't expect this message. Saw this happening a couple of times, but developing these test cases of valid but unexpected messages is a whole sciente and a herculean creativity effort by itself. I don't like the pokemon catch-'em-all, but I'd sleep better knowing that it'll at least not crash the whole thing and put user's funds at risk.
I'll remove it and create an issue for that.